src/troubleshooting.rst

```diff
@@ -0,0 +1,482 @@
+
+Means that ``ssh`` is not installed.
+
+See :ref:`non-python-requirements` for details on system requirements.
+
+
+.. _troubleshooting.remote_init_did_not_complete:
+
+``platform: <name> - initialisation did not complete``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This means that Cylc was unable to install the required workflow files onto
+a remote platform.
+
+This either means that:
+
+1. The platform is down (e.g. all login nodes are offline).
+2. There is a network problem (e.g. you cannot connect to the login nodes).
+3. The platform is not correctly configured.
+
+Check the scheduler log, you might find some stderr associated with this
+message.
+
+If your site has configured this platform for you, it's probably (1) or (2),
+check you are able to access the platform and notify the site administrator as
+appropriate.
+
+If you are in the progress of setting up a new platform, it's probably (3).
+You might want to check that you've configured the
+:cylc:conf:`global.cylc[platforms][<platform name>]install target` correctly,
+note that this defaults to the platform name if not explicitly set.
+
+
+``OperationalError: disk I/O error``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This means that something was unable to write to a file when it would expect to
+have been able to.
+
+This error usually occurs if when you have exceeded you filesystem quota.
+
+If a Cylc workflow cannot write to the filesystem, it will shutdown. Once
+you've cleared out enough space for the workflow to continue you should be able
+to safely restart it as you would normally using ``cylc play``, the workflow
+will continue where it left off.
+
+
+``socket.gaierror``
+^^^^^^^^^^^^^^^^^^^
+
+This usually means that a host could not be found on the network. The likely
+cause is DNS configuration.
+
+Cylc is a distributed system so needs to be able to identify the hosts it has
+been configured to use (e.g. the servers where you run Cylc workflows or any
+login nodes you submit jobs to).
+Cylc expects each host to have a unique and stable fully qualified domain name
+(FQDN) and to be identifiable from other hosts on the network using this name.
+
+I.E. If a host identifies itself with an FQDN, then we should be able to look it
+from another host by this FQDN. If we can't, then Cylc can't tell which host is
+which and will not be able to function properly.
+
+If the FQDN of a host is reported differently from different hosts on the
+network, then Cylc commands will likely fail. To fix the issue, ensure that the
+DNS setup is consistent.
+
+Sometimes we do not have control over the platforms we use and it is not
+possible to compel system administrators to address these issues. If this is
+the case, you can fall back to IP address based host identification which may
+work (i.e. use IP address rather than host names, makes logs less human
+readable). As a last resort you can also hard-code the host name for each host.
```

Comment:
```suggestion
work (i.e. use IP addresses rather than host names, which makes logs less 
readable). As a last resort you can also hard-code the host name for each host